src/QATools/Example/Pages/LoginPage.php

```diff
@@ -0,0 +1,75 @@
+<?php
+/**
+ * This file is part of the QA-Tools library.
+ * For the full copyright and license information, please view
+ * the LICENSE file that was distributed with this source code.
+ *
+ * @copyright Michael Iwersen <mi.iwersen@gmail.com>
+ * @link      https://github.com/qa-tools/behat-example
+ */
+
+namespace QATools\Example\Pages;
+
+
+use QATools\QATools\HtmlElements\Element\Form;
+use QATools\QATools\HtmlElements\TypifiedPage;
+
+/**
+ * Class LoginPage.
+ *
+ * @page-url('/')
+ */
+class LoginPage extends TypifiedPage
+{
+
+	/**
+	 * The login form.
+	 *
+	 * @var Form
+	 * @find-by('id' => 'login-form')
+	 */
+	protected $loginForm;
+
+	/**
+	 * Fill the login form with given data.
+	 *
+	 * @param string $username The username.
+	 * @param string $password The password.
+	 *
+	 * @return static
+	 */
+	public function fillForm($username, $password)
+	{
+		$this->loginForm->fill(array(
+			'username' => $username,
+			'password' => $password,
+		));
+
+		return $this;
+	}
+
+	/**
+	 * Does the login.
+	 *
+	 * @return Page
+	 */
+	public function loginSuccess()
```

Comment:
I thought a bit about scenarios, but too my pity I couldn't get my hands on the tests of my colleagues.

If the submit step gets combined with entering the form the step might be look like `... the user is logging in successful with username and password`. This might be too long for a step and too complex too understand. To make it short it could also be `... the user is logging in successful`, I think this is what the stakeholder would write.

What about all possible cases of failure?
`... the user is logging in not successful with username and password` there are at least three different versions required to cover all. 
By separating the input and submit the code does not need to be repeated multiple time in each step (form input) and can be combined on demand.

```gherkin
Given an user
When entering username xxx
And entering password yyy
Then the login should fail
```

This would allow us to skip password or username input or do a successful login. But maybe thats just because I am too much developer and don't want to repeat myself too often :smile: